### PR TITLE
Largest Binary Search Tree in a Binary Tree

### PR DESCRIPTION
**Pull request by mdnazisharman2803**
- Finding largest Binary Search Tree in a Binary Tree.

## Related Issue
- #617 

## Proposed Changes
- Code to find if a Binary Tree contains Binary Search Tree as well as it's size.
- Readme file for the same.

## Additional Info
- All kind of information is in the readme file as well as comment present in code file wherever necessary.

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [X] 🌟 ed the repo

- I have implemented concept of finding if a Binary Tree  contains Binary Search Tree which is itself a very important concept to understand, as this problem clearly shows that it is not necessary that every Binary Tree  is a Binary Search Tree also this is not necessary that every Binary Tree contains a Binary Search Tree, but this problem shows that as a whole Binary Tree may not be a Binary Search Tree but maybe some of it's portion(sub-tree) can be a Binary Search Tree.

- This concept is not yet implemented and and it solves understanding of Binary Search Tree is simply an organized Binary Tree.

- it fixes issue #617 

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
| Screenshot #1      | Screenshot #2  |
| ----------- | ----------- |
| Code  | Output    |
|Screenshot (478) |  Screenshot (479)|

### CODE REVIEW

File: T-Tree/B-BinaryTree/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+- Call the function bst(root) for the left child and right child of any node, it will tell whether the left/right sub-tree is a BST or not.
+- Add two more data properties in the bstpair class: "root" and "size" which denote the root of the largest BST subtree and size of the largest BST subtree respectively.
+- if node comes to be a bst then
+root node will be : ans.root = node;
+and size will be : ans.size = l.size + r.size + 1;
+1 is added in above expression to include the root node in the "size".
+- Else if the node doesn't form a BST and the size of the left BST subtree of the node is greater than the size of the right BST subtree then the largest BST subtree is the left subtree and its size and root are assigned to "ans".
+- Else the largest BST subtree is the right subtree and its size and root are assigned to "ans".
+
+### Time Complexity
```
Comment (Lakhankumawat) [inline T-Tree/B-BinaryTree/README.md:101]: Please add a working Preview image in your README you forgot last time too so i added one see your README in generic tree you'll get the idea

File: T-Tree/B-BinaryTree/largestBSTinBT.cpp [hunk 2]
```diff
@@ -0,0 +1,131 @@
+#include <bits/stdc++.h>
```
Comment (Lakhankumawat) [inline T-Tree/B-BinaryTree/largestBSTinBT.cpp:1]: File name make L capital in Largest

File: T-Tree/B-BinaryTree/largestBSTinBT.cpp [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+      arr[i] = stoi(tmp);
+    }
+  }
+
+  Node* root = constructTree(arr);
+
+  bst r = Bst(root);
+  cout <<"value of root node of the largest bst subtree is: "<< r.root->data << " and it's size is: " << r.size;
+
+
```
Comment (Lakhankumawat) [inline T-Tree/B-BinaryTree/largestBSTinBT.cpp:129]: Rest looks good to me i'll review your code when you've done these changes

**Review by Lakhankumawat — CHANGES_REQUESTED** [review]

**Comment by mdnazisharman2803** [comment]
@Lakhankumawat done resolved

**Pull request merged**